pkg/apis/operator/v1alpha1/validation/validation.go

```diff
@@ -109,7 +110,24 @@ func validateVirtualClusterUpdate(oldGarden, newGarden *operatorv1alpha1.Garden)
 	// First domain is immutable. Changing this would incompatibly change the service account issuer in the cluster, ref https://github.com/gardener/gardener/blob/17ff592e734131ef746560641bdcdec3bcfce0f1/pkg/component/kubeapiserver/deployment.go#L585C8-L585C8
 	// Note: We can consider supporting this scenario in the future but would need to re-issue all service account tokens during the reconcile run.
 	if len(oldVirtualCluster.DNS.Domains) > 0 && len(newVirtualCluster.DNS.Domains) > 0 {
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(oldVirtualCluster.DNS.Domains[0], newVirtualCluster.DNS.Domains[0], fldPath.Child("dns", "domains").Index(0))...)
+		allErrs = append(allErrs, apivalidation.ValidateImmutableField(oldVirtualCluster.DNS.Domains[0].Name, newVirtualCluster.DNS.Domains[0].Name, fldPath.Child("dns", "domains").Index(0).Child("name"))...)
+	}
+
+	if oldVirtualCluster.ETCD != nil && oldVirtualCluster.ETCD.Main != nil && oldVirtualCluster.ETCD.Main.Backup != nil {
+		if newVirtualCluster.ETCD != nil && newVirtualCluster.ETCD.Main != nil {
+			fldBackup := fldPath.Child("etcd", "main", "backup")
+			if newVirtualCluster.ETCD.Main.Backup != nil {
+				if oldVirtualCluster.ETCD.Main.Backup.BucketName == nil && newVirtualCluster.ETCD.Main.Backup.BucketName != nil {
+					allErrs = append(allErrs, field.Forbidden(fldBackup.Child("bucketName"), "bucket name must remain unset if it was not set before"))
+				}
+				if oldVirtualCluster.ETCD.Main.Backup.BucketName != nil && newVirtualCluster.ETCD.Main.Backup.BucketName == nil {
+					allErrs = append(allErrs, field.Forbidden(fldBackup.Child("bucketName"), "bucket name must remain unchanged if it was set before"))
+				}
+			} else {
+				allErrs = append(allErrs, field.Forbidden(fldBackup, "backup must not be deactivated if it was set before"))
+				allErrs = append(allErrs, apivalidation.ValidateImmutableField(newVirtualCluster.ETCD.Main.Backup, oldVirtualCluster.ETCD.Main.Backup, fldPath.Child("backup"))...)
+			}
```

Comment:
Isn't this too nested and not all cases are covered?

Maybe a check like the following would be clearer:

```
if newVirtualCluster.ETCD != nil || newVirtualCluster.ETCD.Main != nil || newVirtualCluster.ETCD.Main.Backup != nil {
				allErrs = append(allErrs, field.Forbidden(fldBackup, "backup must not be deactivated if it was set before"))
}
```

Reply:
changed as suggested